### ISSUE TITLE
How to use tx iterator?

### ISSUE THREAD

**Issue by lzl124631x** [citp/BlockSci#186]
The [doc](https://citp.github.io/BlockSci/reference/chain/iterators/TxIterator.html) doesn't guide me how to use it. I want to iterate through the txs and run some custom logic. Thanks

```
in: blocksci.TxIterator.all
out: <property at 0x7f156433e638>
```

**Comment by maltemoeser** [comment]
Certain functions may return a TxIterator, such as
```
addr = chain.address_from_string("1BTCorgHwCg6u2YSAWKgS17qUad6kHmtQW")
addr.out_txes
> <blocksci.TxIterator at 0x7ef1d81ca620>
```

You wouldn't instantiate a TxIterator yourself.

**Comment by lzl124631x** [comment]
![image](https://user-images.githubusercontent.com/5824517/47542466-6409f880-d892-11e8-9721-85b00244bd16.png)

**Comment by maltemoeser** [comment]
Ah sorry about the example, the API might have changed between 0.5 and the upcoming 0.6 that I'm using locally.

To go through all transactions, you can use `chain.blocks.txes`, which will return a `TxIterator`

**Comment by lzl124631x** [comment]
@maltemoeser  Thanks! I've tried that but that command seems to run forever (Maybe I need to wait longer)

**Comment by maltemoeser** [comment]
Depending on what you're doing, it can take a while, you're iterating over and instantiating hundreds of millions of transactions. The next BlockSci version will have some new tools to bring the Python interface closer to the C++ performance. Until then, if you want, I'm happy to take a look at your particular code example to see if there's any low-hanging performance improvements you could do

**Comment by lzl124631x** [comment]
@maltemoeser  I want to translate bitcoin tx to a tx of the blockchain I'm developing. 

I wrote a golang program to traverse the bitcoin txes using btcd RPC. But the best TPS I can get is just roughly ~1000. (If the overhead of transforming the txes is took into consideration, I can just get ~666 TPS). It's too slow for me.

There are at least 350m bitcoin txes. If the TPS is 1k, it will take 243 days to traverse all the txes :(

That's why I started to look into Blocksci hoping to get higher TPS. But even with the dummy `visit` method (just pushing the tx into an array) the TPS I could get is just 600+. I know python is relatively slow, but is there anyway to make it faster? I can write C++, but want to know the potential before invest more time into it.
![image](https://user-images.githubusercontent.com/5824517/47610222-d4388b80-da04-11e8-98ea-220861fafba2.png)
